src/main/java/com/evgeniyfedorchenko/hogwarts/services/FacultyServiceImpl.java

```diff
@@ -112,4 +113,25 @@ public List<StudentOutputDto> findStudents(Long id) {
                 .map(studentMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public Optional<String> findLongestName() {
+
+        AtomicReference<String> longestName = new AtomicReference<>("");
+
+        logger.debug("Invoke non-optimal method \"facultyRepository.findAll()\"");
+        facultyRepository.findAll().stream()
+                .map(Faculty::getName)
+                .forEach(name -> {
+                    if (longestName.get().length() < name.length()) {
+                        longestName.set(name);
+                    }
+                });
+        return longestName.get().equals("") ? Optional.empty() : Optional.of(longestName.get());
```

Comment:
а зачем так сложно? почему бы не так?
```suggestion
        return facultyRepository.findAll().stream()
                .map(Faculty::getName)
                .max(String::length);
```

Reply:
Воображения не хватило :D